pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -83,10 +83,10 @@ func (pl *NodeAffinity) EventsToRegister() []framework.ClusterEvent {
 }
 
 // PreFilter builds and writes cycle state used by Filter.
-func (pl *NodeAffinity) PreFilter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod) *framework.Status {
+func (pl *NodeAffinity) PreFilter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
```

Comment:
Are you going to make the DS change in a follow-up PR?

Reply:
yeah, I thought we first get the interface change first and then do one specific to NodeAffinity.

Reply:
what if pluginA returns {node1, node2}, while pluginB returns {node3, node4}. Let's suppose their logic are both correct, in this case the merging result is an empty obj, and then we're going to iterate all nodes?

Reply:
In the current implementation, yes, that is what we will end up with unless we want `RunPreFilterPlugins` to detect the case and return an unschedulable status, which I think isn't really worth it.

I am concerned about distinguishing between nil and empty, we already have status that gives us a clear indication whether or not to continue downstream.